src/libs/portfolio/helpers.ts

```diff
@@ -262,42 +263,203 @@ export const mapToken = (
+      erc20.decimals().catch((e) => handleERC20Error(e, 'decimals'))
+    ])
+  } catch (e) {
+    handleERC20Error(e, 'token validation')
+  }
 
   if (
     typeof balance === 'undefined' ||
     typeof symbol === 'undefined' ||
     typeof decimals === 'undefined'
   ) {
-    isValid = false
+    // Only mark as invalid if it's not a network error
+    if (!hasNetworkError) {
+      isValid = false
+      if (!errorMessage) {
+        errorMessage = 'Token validation failed: unable to fetch required token data'
+        errorType = 'validation'
+      }
+    }
+  } else if (!hasNetworkError) {
+    // Reset error state only if validation succeeded AND there was no network error
+    isValid = true
+    errorMessage = ''
+    errorType = null
   }
 
-  isValid = isValid && !hasError
-  return [isValid, type]
+  // If validation failed and network detection is enabled, check other networks
+  if (!isValid && !hasNetworkError && enableNetworkDetection && allNetworks && allProviders) {
+    try {
+      // Get candidate networks and limit the number to check
+      const candidateNetworks = allNetworks
+        .filter((network) => allProviders[network.chainId.toString()]?.isWorking !== false)
+        .filter((network) => network.chainId !== token.chainId) // Skip the current network
+        .slice(0, maxNetworksToCheck) // Limit the number of networks to check
+
+      // Use concurrency-limited validation to prevent overwhelming RPC providers
+      const validationFunction = async (network: Network) => {
+        try {
+          const networkProvider = allProviders[network.chainId.toString()]
+          if (!networkProvider) return null
+
+          // Use validateERC20Token without network detection to avoid circular dependency
+          const validation = await validateERC20Token(
+            { address: token.address, chainId: network.chainId },
+            accountId,
+            networkProvider,
+            { enableNetworkDetection: false }
+          )
+
+          return validation.isValid ? network : null
+        } catch (error) {
+          return null
+        }
+      }
+
+      const results = await limitConcurrency(
+        candidateNetworks,
+        validationFunction,
+        concurrencyLimit
+      )
+      const validNetworks = results.filter((network): network is Network => network !== null)
+
+      if (validNetworks.length > 0) {
+        const networkNames = validNetworks.map((net) => net.name).join(', ')
+        errorMessage = `This token is found on ${networkNames}. Is the correct network selected?`
+        errorType = 'validation'
+      }
+    } catch (networkDetectionError) {
+      // Network detection failed, but don't override the original error
+      console.warn('Network detection failed:', networkDetectionError)
+    }
+  }
+
+  return {
+    isValid,
+    standard: type,
+    error: {
+      message: errorMessage || null,
+      type: errorType
+    }
```

Comment:
Error message and error type go hand in hand - it should be an object as a variable and its default vault should be null.

```
errorMessage = `This token is found on ${networkNames}. Is the correct network selected?`
errorType = 'validation'
```

To:
```
error = { type: ..., message: ... }
```